pkg/lockfile/parse-pdm-lock_test.go

```diff
@@ -0,0 +1,198 @@
+package lockfile_test
+
+import (
+	"io/fs"
+	"testing"
+
+	"github.com/google/osv-scanner/pkg/lockfile"
+)
+
+func TestPdmExtractor_ShouldExtract(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name string
+		path string
+		want bool
+	}{
+		{
+			name: "empty",
+			path: "",
+			want: false,
+		},
+		{
+			name: "plain",
+			path: "pdm.lock",
+			want: true,
+		},
+		{
+			name: "absolute",
+			path: "/path/to/pdm.lock",
+			want: true,
+		},
+		{
+			name: "relative",
+			path: "../../pdm.lock",
+			want: true,
+		},
+		{
+			name: "in-path",
+			path: "/path/with/pdm.lock/in/middle",
+			want: false,
+		},
+		{
+			name: "invalid-suffix",
+			path: "pdm.lock.file",
+			want: false,
+		},
+		{
+			name: "invalid-prefix",
+			path: "project.name.pdm.lock",
+			want: false,
+		},
+	}
+
+	for _, test := range tests {
+		tst := test
+		t.Run(tst.name, func(t *testing.T) {
+			t.Parallel()
+			ext := lockfile.PdmLockExtractor{}
+			should := ext.ShouldExtract(tst.path)
+			if should != tst.want {
+				t.Errorf("ShouldExtract() - got %v, expected %v", should, tst.want)
+			}
+		})
+	}
+}
+
+func expectNilErr(t *testing.T, err error) {
+	t.Helper()
+	if err != nil {
+		t.Errorf("got unexpected error: %v", err)
+	}
+}
+
+func TestParsePdmLock_FileDoesNotExist(t *testing.T) {
+	t.Parallel()
```

Comment:
nit: these are pretty squashed together

usually I like to have a new line between every function, after `t.Parallel()`, and before the first `expect*` call

Reply:
Added newlines to test functions.